lightning/src/onion_message/blinded_route.rs

```diff
@@ -113,6 +114,52 @@ fn encrypt_payload<P: Writeable>(payload: P, encrypted_tlvs_ss: [u8; 32]) -> Vec
 	writer.0
 }
 
+impl Writeable for BlindedRoute {
+	fn write<W: Writer>(&self, w: &mut W) -> Result<(), io::Error> {
+		self.introduction_node_id.write(w)?;
+		self.blinding_point.write(w)?;
+		for hop in &self.blinded_hops {
+			hop.write(w)?;
+		}
+		Ok(())
+	}
+}
+
+impl Readable for BlindedRoute {
+	fn read<R: io::Read>(r: &mut R) -> Result<Self, DecodeError> {
+		let introduction_node_id = Readable::read(r)?;
+		let blinding_point = Readable::read(r)?;
+		let mut blinded_hops: Vec<BlindedHop> = Vec::new();
+		loop {
+			let mut tracking_reader = ReadTrackingReader::new(&mut *r);
+			let blinded_hop: BlindedHop = match Readable::read(&mut tracking_reader) {
+				Ok(h) => h,
+				Err(DecodeError::ShortRead) => {
+					if !tracking_reader.have_read {
+						break
+					}
+					return Err(DecodeError::ShortRead)
+				},
+				Err(e) => return Err(e)
+			};
+			blinded_hops.push(blinded_hop);
```

Comment:
This can be simplified to:

```rust
match Readable::read(&mut tracking_reader) {
    Ok(h) => blinded_hops.push(h),
    Err(DecodeError::ShortRead) if !tracking_reader.have_read => break,
    Err(e) => return Err(e),
}
```